### ISSUE TITLE
Unable to use a custom theme from a jar / submodule

### ISSUE THREAD

**Issue by jcgueriaud1** [vaadin/flow#12247]
### Description of the bug / feature

I want to package a theme as a reusable theme in a jar and use it in a Vaadin application.
The setup I tried is a multimodule Maven module with Spring boot. (I also tried with jetty with the same result)

### Minimal reproducible example

Get the application:
https://github.com/jcgueriaud1/multimodule-spring-vaadin-app/tree/feature/vaadin_theme

Run: 
`mvn clean install -PproductionMode`

>  Discovered @Theme annotation with theme name 'reusable-theme', but could not find the theme directory in the project or available as a jar dependency. Check if you forgot to create the folder under '/Users/jean-christophe/Documents/vaadin-components-factory/multimodule-spring-vaadin-app/ui-bootstrap/frontend/themes' or have mistyped the theme or folder name for 'reusable-theme'. 

I also tried to run it in dev mode with the same result.

I tried to keep the theme as close as possible with the flow-tests.

### Expected behavior

The theme should be available and the application should work.

### Actual behavior
See the error.

### Versions:
    - Vaadin / Flow version: V14.7.3
    - Java version: 8 and 14
    - OS version: Mac

**Comment by jcgueriaud1** [comment]
I also try with Vaadin 21.0.3 in this branch: https://github.com/jcgueriaud1/multimodule-spring-vaadin-app/tree/feature/vaadin_theme_21

I don't have any error when I start the application.
In development it seems to work (I didn't try everything).
![image](https://user-images.githubusercontent.com/51313578/139871612-8233b41a-5a25-48dd-94c7-c6749fdd7b97.png)

In production mode it does not work completely. (the font and the background images are not working but the h1 is red)
![image](https://user-images.githubusercontent.com/51313578/139871797-eb4adccf-51b4-4648-9af5-2cf6347b438b.png)

![image](https://user-images.githubusercontent.com/51313578/139871771-78f115e5-ba8c-44f2-a0a1-cdca06a7d823.png)

**Comment by caalador** [comment]
The issue as seen in the given application is that the reusable theme resources are in the wrong place.
In the given sample the theme resources are in `src/main/resources/META-INF/resources/frontend/themes/reusable-theme` when they should be in `src/main/resources/META-INF/resources/themes/reusable-theme`.

The path in the given sample sets the files to `target/frontend/frontend/themes` where as the place the theme is searched from is `target/frontend/themes`.

For more information see [Packaging a Theme for Reuse](https://vaadin.com/docs/v14/flow/styling/custom-theme-packaging)

**Comment by jcgueriaud1** [comment]
That explains a lot of things :). Thanks for spotting this, I'm so used to put everything in the frontend folder I didn't notice the error.

I've updated my project in Vaadin 21 and 14 and both are working now.

I've noticed that a VAADIN folder is created at the root of the folder when I run the project in Vaadin 14.

What's the purpose of this folder and why it's in the project root folder?

Steps to reproduce:
- get the project: https://github.com/jcgueriaud1/multimodule-spring-vaadin-app/tree/feature/vaadin_theme
- run `mvn clean install` in the root folder
- go the ui-bootstrap folder
- run `mvn spring-boot:run`
![image](https://user-images.githubusercontent.com/51313578/140020193-2099aaa2-d7f8-481d-bc00-89beadee2c54.png)
